### DISCUSSION TITLE
Using the simple persistence within Next.js v12

### DISCUSSION THREAD

**Discussion by maxcell** [pmndrs/jotai#2176] [Q&A] [answered]
Howdy! First time poster here!

I have read over the documentation, discussions, and issues before coming here to ask the question. I am curious for your example you have for the [simple pattern with localStorage](https://jotai.org/docs/guides/persistence#a-simple-pattern-with-localstorage). I was attempting to use Jotai v1.7.6 in a Next.js v12 project and I realized there would be some issues with gathering the persisted value due to SSR. I assume that's why there is an issue with reading the window object like in this screenshot:

![CleanShot 2023-10-12 at 09 57 42](https://github.com/pmndrs/jotai/assets/8431042/970eb108-382e-451e-8ba8-4b6c60855775)

Would it mean that all components that attempt to read the persisted atom would need to essentially use the client-only wrapper you have shared before? I believe here https://jotai.org/docs/utilities/storage#server-side-rendering. My first goal was to attempt to use `atomWithStorage` but it looks like that will also stringify the value so if you want to store a string this makes `sessionStorage`'s value become `'"some string"'` so that's why I was going down this approach.

I am hoping the answer is there is another way but if not it is okay! Thank you in advance. I created a small reproduction here https://stackblitz.com/edit/nextjs-uxttdo?file=pages%2Findex.tsx

**Comment by dai-shi** [comment, marked as answer]
> Would it mean that all components that attempt to read the persisted atom would need to essentially use the client-only wrapper you have shared before?

Yeah, because it can't read client things on server.

Or, you can use the default value on server:
```js
const strAtom = atom(typeof window !== 'undefined' ? window.sessionStorage.getItem(key) : initialValue);
```

**Reply by maxcell** [reply]
That makes sense! Thank you for your suggestion!
